src/components/shared/TreeView.js

```diff
@@ -520,50 +534,83 @@ class TreeView<
+          if (event.metaKey) {
+            // On MacOS this is a common shortcut for the Home gesture
+            this._select(visibleRows[0]);
+            break;
+          }
+
+          if (selectedRowIndex > 0) {
+            this._select(visibleRows[selectedRowIndex - 1]);
           }
           break;
-        case 38: // KEY_UP
-          {
-            if (selectedRowIndex > 0) {
-              this._select(visibleRows[selectedRowIndex - 1]);
-            }
+        }
+        case 'ArrowDown': {
+          if (event.metaKey) {
+            // On MacOS this is a common shortcut for the End gesture
+            this._select(visibleRows[visibleRows.length - 1]);
+            break;
+          }
+
+          if (selectedRowIndex < visibleRows.length - 1) {
+            this._select(visibleRows[selectedRowIndex + 1]);
+          }
+          break;
+        }
+        case 'PageUp': {
+          if (selectedRowIndex > 0) {
+            const nextRow = Math.max(0, selectedRowIndex - PAGE_KEYS_DELTA);
+            this._select(visibleRows[nextRow]);
           }
           break;
-        case 39: // KEY_RIGHT
-          {
-            const isCollapsed = this._isCollapsed(selected);
-            if (isCollapsed) {
-              this._toggle(selected);
-            } else {
-              // Do KEY_DOWN only if the next element is a child
-              if (this.props.tree.hasChildren(selected)) {
-                this._select(this.props.tree.getChildren(selected)[0]);
-              }
+        }
+        case 'PageDown': {
+          if (selectedRowIndex < visibleRows.length - 1) {
+            const nextRow = Math.min(
+              visibleRows.length - 1,
+              selectedRowIndex + PAGE_KEYS_DELTA
+            );
+            this._select(visibleRows[nextRow]);
+          }
+          break;
+        }
+        case 'Home': {
+          this._select(visibleRows[0]);
+          break;
+        }
+        case 'End': {
+          this._select(visibleRows[visibleRows.length - 1]);
+          break;
+        }
+        case 'ArrowLeft': {
+          const isCollapsed = this._isCollapsed(selected);
+          if (!isCollapsed) {
+            this._toggle(selected);
+          } else {
+            const parent = this.props.tree.getParent(selected);
+            if (parent !== -1) {
+              this._select(parent);
             }
           }
           break;
-        case 40: // KEY_DOWN
-          {
-            if (selectedRowIndex < visibleRows.length - 1) {
-              this._select(visibleRows[selectedRowIndex + 1]);
+        }
+        case 'ArrowRight': {
+          const isCollapsed = this._isCollapsed(selected);
+          if (isCollapsed) {
+            this._toggle(selected);
+          } else {
+            // Do KEY_DOWN only if the next element is a child
+            if (this.props.tree.hasChildren(selected)) {
+              this._select(this.props.tree.getChildren(selected)[0]);
             }
           }
```

Comment:
I reorganized the switch to organize it by pair (eg: ArrowUp close to ArrowDown)